src/base/SubscriptionGateway.php

```diff
@@ -527,7 +527,10 @@ public function handlePaymentIntentSucceeded(array $data): void
             $updateTransaction = null;
 
             if ($transaction->parentId === null) {
```

Comment:
I think just doing 

`if (transaction && $transaction->parentId === null) {`

should be enough